pkgs/by-name/ja/jazz2/package.nix

```
@@ -12,36 +12,40 @@
 , stdenv
 , xorg
 , zlib
+, graphicsLibrary ? "GLFW"
 }:
 
-stdenv.mkDerivation rec {
+assert lib.assertOneOf "graphicsLibrary" graphicsLibrary [ "SDL2" "GLFW" ];
+stdenv.mkDerivation (finalAttrs: {
   pname = "jazz2";
-  version = "2.1.0";
+  version = "2.2.1";
 
   src = fetchFromGitHub {
     owner = "deathkiller";
     repo = "jazz2-native";
-    rev = version;
-    sha256 = "nJha7+geP2Ov7ciEDzJ+XWdiF1jzv4Oeis1DwxcpJXo=";
+    rev = finalAttrs.version;
+    hash = "sha256-+N7yhp/To5Ic9Zejc/ge37kkdPdQC5h5a1FqIPRDtLQ=";
   };
 
   patches = [ ./nocontent.patch ];
 
-  buildInputs = [ libGL SDL2 zlib glew glfw openal libvorbis libopenmpt xorg.libSM xorg.libICE xorg.libXext ];
+  buildInputs = [ glew glfw libGL libopenmpt libvorbis openal SDL2 xorg.libICE xorg.libSM xorg.libXext zlib ];
```

I think if the graphics backend of choice is selectable at build-time now, then only pulling in the dependencies for the selected backend makes sense?

```suggestion
  buildInputs = [
    # common stuff used by all backends
  ] ++ lib.optionals (graphicsLibrary == "GLFW") [
    # glfw-only stuff, i.e. glfw
  ] ++ lib.optionals (graphicsLibrary == "SDL2") [
    # SDL-only stuff, i.e. SDL2
  ];
```

---

sure 1 sec